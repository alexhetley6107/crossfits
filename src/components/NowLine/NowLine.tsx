import { theme } from '@/themes/theme';
import React from 'react';
import styled from 'styled-components';

export type NowLineProps = {
  text: string;
  top?: number;
};

const NowLineWrapper = styled.div<{ $top: number }>`
  position: absolute;
  height: 1.5px;
  background-color: ${theme.colors.primary};
  width: 100%;
  right: 0;
  left: 0;
  top: ${({ $top }) => $top}px;
`;

const StyledText = styled.span`
  position: absolute;
  right: 0;
  bottom: 5px;
  color: ${theme.colors.primary};
  font-size: 14px;
`;

export const NowLine: React.FC<NowLineProps> = ({ text, top = 0 }) => {
  return (
    <NowLineWrapper $top={top} data-testid="NowLine">
      <StyledText>{text}</StyledText>
    </NowLineWrapper>
  );
};
