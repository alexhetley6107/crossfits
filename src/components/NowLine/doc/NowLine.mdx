import { Canvas, Meta } from '@storybook/blocks';
import { Example } from './Example.tsx';
import * as NowLine from './NowLine.stories.tsx';

<Meta of={NowLine} title="NowLine" />

# Button

<Canvas of={NowLine.Base} />

## Usage

```ts
import { NowLine } from 'remindee-ui-kit';

export const Example = () => {
  return <NowLine text="Now" />;
};
```

### Arguments

- **text** - A string value indicating text message over blue line.
- **top** - A number value which show where NowLine will be placed.
- **className** - A string that sets css class to the root of component.
