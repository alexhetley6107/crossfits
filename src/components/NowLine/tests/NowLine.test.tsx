import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { NowLine } from '../NowLine';
import '@testing-library/jest-dom';
import 'jest-styled-components';

describe('NowLine component', () => {
  describe('General', () => {
    it('Render', () => {
      render(<NowLine text="Now" />);
      const target = screen.getByTestId('NowLine');
      expect(target).toBeInTheDocument();
    });
  });
  describe('Props', () => {
    it('Text', () => {
      render(<NowLine text="Now" />);
      const text = screen.getByText('Now');
      expect(text).toBeInTheDocument();
    });
    it('Top', () => {
      render(<NowLine text="Now" top={35} />);
      const target = screen.getByTestId('NowLine');
      expect(target).toMatchSnapshot();
    });
  });
});
