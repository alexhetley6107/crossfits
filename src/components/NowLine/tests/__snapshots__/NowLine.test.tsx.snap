// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NowLine component > Props > Top 1`] = `
.c0 {
  position: absolute;
  height: 1.5px;
  background-color: #346beb;
  width: 100%;
  right: 0;
  left: 0;
  top: 35px;
}

.c1 {
  position: absolute;
  right: 0;
  bottom: 5px;
  color: #346beb;
  font-size: 14px;
}

<div
  class="c0"
  data-testid="NowLine"
>
  <span
    class="c1"
  >
    Now
  </span>
</div>
`;
